### PR TITLE
Handle missing TERM variable. Fixes #145

### LINKED ISSUE #145: panic on multirust update (on jenkins)
While trying to setup a jenkins build job for my rust project I encounter a panic of multirust.
```
$ multirust update nightly
thread '<main>' panicked at 'Failed to open terminal', ../src/libcore/option.rs:704
stack backtrace:
   1:     0x7f4ca6c6bda0 - sys::backtrace::tracing::imp::write::hef3f5235812bde76Qwu
   2:     0x7f4ca6c737bb - panicking::default_handler::_$u7b$$u7b$closure$u7d$$u7d$::closure.43471
   3:     0x7f4ca6c73428 - panicking::default_handler::hb72e26fd774dcea5E3y
   4:     0x7f4ca6c59e0c - sys_common::unwind::begin_unwind_inner::hcccaf464c9787883Flt
   5:     0x7f4ca6c5a438 - sys_common::unwind::begin_unwind_fmt::h522f8632783201c5Lkt
   6:     0x7f4ca6c6b151 - rust_begin_unwind
   7:     0x7f4ca6ca342f - panicking::panic_fmt::hd3ab350bce4886e96YL
   8:     0x7f4ca6caa454 - option::expect_failed::h7a80286ec067e9d5ftO
   9:     0x7f4ca6aaa314 - common::set_globals::h527996c7050c2e00bqa
  10:     0x7f4ca6a9ccb7 - multirust_mode::main::h6faae5d83432a340p1a
  11:     0x7f4ca6a8ff9f - main::h278b7db25a676f57Qhd
  12:     0x7f4ca6c73084 - sys_common::unwind::try::try_fn::h5713377621479519318
  13:     0x7f4ca6c6b0db - __rust_try
  14:     0x7f4ca6c72b0e - rt::lang_start::hf9917a10f7f86c4fKVy
  15:     0x7f4ca57dbb14 - __libc_start_main
  16:     0x7f4ca6a8faa8 - <unknown>
  17:                0x0 - <unknown>
```
The source seems to be [this](https://github.com/rust-lang-nursery/multirust-rs/blob/master/src/multirust-cli/download_tracker.rs#L48) line.
As workaround for this I set added 
```bash
export TERM=screen
``` 
to my build script.

### PR DESCRIPTION
**Pull request by brson**
This creates some ugly and rarely-executed branches.

### CODE REVIEW

**Comment by alexcrichton** [comment]
Hm this is another situation where [something in Cargo is very similar](https://github.com/rust-lang/cargo/blob/4e6434dab92fe56371036c37bddb49a87e621504/src/cargo/core/shell.rs#L144). Would it be possible to refactor this with a shared type that doesn't have these duplication problems? That way you'd say "set the color to blue" and it'd just ignore it if `TERM` couldn't get parsed for whatever reason.

**Comment by brson** [comment]
Well I was hoping you wouldn't say that, but yes.

**Comment by brson** [comment]
Updated to wrap `StdoutTerminal` and `StderrTerminal` in a type that deals with missing terminfo.

File: src/multirust-cli/download_tracker.rs [hunk 1]
```diff
@@ -144,7 +145,7 @@ impl DownloadTracker {
                 } else {
                     Cow::Borrowed("Unknown")
                 };
-                let _ = write!(&mut self.term,
+                let _ = write!(self.term.as_mut().unwrap(),
```
Comment (alexcrichton) [inline src/multirust-cli/download_tracker.rs]: Won't these unwraps trigger like the original unwrap this was trying to remove?

Comment (brson) [inline src/multirust-cli/download_tracker.rs]: No. There is a branch in `handle_notification` that short-circuits most of the code in this module.

Comment (alexcrichton) [inline src/multirust-cli/download_tracker.rs]: Aha! sounds good to me

**Pull request merged**